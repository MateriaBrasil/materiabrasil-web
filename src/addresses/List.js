import React from 'react'
import { List } from 'croods'

import renderList from './renderList'

export default props => {
  const { id } = props

  return (
    <List
      parentId={id}
      name="addresses"
      path={`/suppliers/${id}/addresses`}
      render={renderList(props)}
    />
  )
}
