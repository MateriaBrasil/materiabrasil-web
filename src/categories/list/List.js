import React, { Component, Fragment } from 'react'
import Typography from '@material-ui/core/Typography'
import isEmpty from 'lodash/isEmpty'
import Button from '@material-ui/core/Button'

import fullName from 'categories/fullName'
import renderCategory from 'categories/renderCategory'
import handleChangeExpanded from 'categories/handleChangeExpanded'

export default class extends Component {
  constructor(props) {
    super(props)
    this.state = { expandedCategories: [] }
    this.onChangeExpanded = handleChangeExpanded.bind(this)
  }

  render() {
    const { list, filters } = this.props
    const { selectedCategories } = filters

    const props = {
      ...this.props,
      ...this.state,
      onChangeExpanded: this.onChangeExpanded,
    }
    console.log(selectedCategories, this.props)

    return (
      <Fragment>
        {!isEmpty(selectedCategories) && (
          <div style={{ marginBottom: 30 }}>
            <Typography variant="subheading" style={{ marginBottom: 20 }}>
              Filtros selecionados
            </Typography>
            {selectedCategories.map(category => {
              console.log('category:',category)
              return (
                <Typography key={category.id} variant="body1">
                  {fullName(list, category, category.name)}
                </Typography>
              )
            })}
            <Button
              style={{ marginTop: 20 }}
              variant="outlined"
              onClick={filters.actions.reset}
            >
              Limpar filtros
            </Button>
          </div>
        )}
        {list.map(renderCategory(props))}
      </Fragment>
    )
  }
}
