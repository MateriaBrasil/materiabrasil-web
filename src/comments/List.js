import React from 'react'
import { List } from 'croods'

import renderList from './renderList'

export default props => {
  const { id, type } = props

  return (
    <List
      parentId={id}
      name="comments"
      path={`/${type}/${id}/comments`}
      render={renderList(props)}
    />
  )
}
