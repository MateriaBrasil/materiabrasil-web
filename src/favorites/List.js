import React from 'react'
import { List } from 'croods'

import renderList from './renderList'

export default props => {
  const { id, parentId } = props

  return (
    <List
      parentId={parentId}
      name="favorites"
      path={`/albums/${id}/favorites`}
      render={renderList(props)}
    />
  )
}
