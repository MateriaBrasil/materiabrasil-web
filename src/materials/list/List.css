.materials-list-grid {
  margin-top: 10px !important;
}

@media (min-width: 960px)
{
  .materials-list-grid {
    margin-top: 20px !important;
  }
}

@media (min-width: 1280px)
{
  .materials-list-grid {
    margin-top: -40px !important;
  }
}
