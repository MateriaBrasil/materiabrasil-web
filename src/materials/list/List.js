import React, { Component } from 'react'
import isEmpty from 'lodash/isEmpty'
import Grid from '@material-ui/core/Grid'

import Error from 'Error'
import renderIcons from './renderIcons'
import Card from './Card'
import './List.css'

export default class extends Component {
  componentDidUpdate(prevProps) {
    const { location, listProps } = this.props
    const { location: prevLocation } = prevProps

    if (listProps && location.search !== prevLocation.search) {
      const { actions } = listProps
      actions.fetchList()
    }
  }

  render() {
    const { list, currentUser } = this.props

    if (isEmpty(list)) {
      return <Error color="default">Nenhum material encontrado.</Error>
    }

    return (
      <Grid item xs={12} md={9} className="materials-list-grid">
        <Grid container spacing={8}>
          {list.map(material => {
            const { id } = material
            return (
              <Card
                key={id}
                material={material}
                renderIcons={renderIcons({ currentUser, id, material })}
                {...this.props}
              />
            )
          })}
        </Grid>
      </Grid>
    )
  }
}
