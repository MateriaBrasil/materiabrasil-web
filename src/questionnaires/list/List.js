import React from 'react'
import map from 'lodash/map'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import FormGroup from '@material-ui/core/FormGroup'
import FormControl from '@material-ui/core/FormControl'
import RadioGroup from '@material-ui/core/RadioGroup'
import Radio from '@material-ui/core/Radio'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Typography from '@material-ui/core/Typography'

export default ({ list }) => (
  <List>
    {map(list, questionnaire => {
      const { name, questions } = questionnaire

      return (
        <ListItem key={name}>
          <Typography variant="headline">{name}</Typography>
          {map(questions, question => {
            const { options, description } = question

            return (
              <FormGroup key={description}>
                <Typography variant="subheading">{description}</Typography>
                <FormControl component="fieldset">
                  <RadioGroup
                    key={description}
                    aria-label="Questionnarie Options"
                    name="questionnaire-options"
                  >
                    {map(options, option => (
                      <FormControlLabel
                        value={toString(option.id)}
                        key={option.id}
                        disabled
                        control={<Radio />}
                        label={option.description}
                      />
                    ))}
                  </RadioGroup>
                </FormControl>
              </FormGroup>
            )
          })}
        </ListItem>
      )
    })}
  </List>
)
