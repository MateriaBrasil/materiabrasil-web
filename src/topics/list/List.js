import React from 'react'
import Link from 'react-router-dom/Link'
import map from 'lodash/map'
import Card from '@material-ui/core/Card'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import Avatar from '@material-ui/core/Avatar'
import PersonIcon from '@material-ui/icons/Person'

import moment from '../../moment'

export default ({ list }) => (
  <Card>
    <List>
      {map(list, topic => {
        const { id, user, subject, createdAt } = topic
        const { firstName, lastName } = user
        const name = `${firstName} ${lastName}`

        return (
          <Link to={`/forum/${id}`} style={{ textDecoration: 'none' }} key={id}>
            <ListItem button>
              <Avatar>
                <PersonIcon />
              </Avatar>
              <ListItemText
                primary={subject}
                secondary={`Criado por ${name} hÃ¡ ${moment(
                  createdAt,
                ).fromNow()}`}
              />
            </ListItem>
          </Link>
        )
      })}
    </List>
  </Card>
)
