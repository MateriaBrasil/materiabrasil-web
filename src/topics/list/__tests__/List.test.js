import React from 'react'
import renderer from 'react-test-renderer'
import List from '../List'

jest.mock('react-router-dom/Link', () => props => (
  <div {...props}>{props.children}</div>
))

const list = [
  {
    id: 123,
    user: {
      id: 111,
      firstName: 'Foo',
      lastName: 'Bar',
    },
    subject: 'Foo subject',
    text: 'Foo text',
  },
  {
    id: 321,
    user: {
      id: 222,
      firstName: 'Bar',
      lastName: 'Foo',
    },
    subject: 'Bar subject',
    text: 'Bar text',
  },
]

it('renders correctly', () => {
  const tree = renderer.create(<List list={list} />).toJSON()
  expect(tree).toMatchSnapshot()
})
