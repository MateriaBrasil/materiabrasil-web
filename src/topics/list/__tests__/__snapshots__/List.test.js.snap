// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1"
>
  <ul
    className="MuiList-root-29 MuiList-padding-30"
  >
    <div
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/forum/123"
    >
      <div
        className="MuiButtonBase-root-43 MuiListItem-root-33 MuiListItem-default-36 MuiListItem-gutters-40 MuiListItem-button-41"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiAvatar-root-46 MuiAvatar-colorDefault-47"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-49"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </g>
          </svg>
        </div>
        <div
          className="MuiListItemText-root-55"
        >
          <h3
            className="MuiTypography-root-61 MuiTypography-subheading-68 MuiListItemText-primary-58"
          >
            Foo subject
          </h3>
          <p
            className="MuiTypography-root-61 MuiTypography-body1-70 MuiTypography-colorTextSecondary-83 MuiListItemText-secondary-59"
          >
            Criado por Foo Bar h치 segundos atr치s
          </p>
        </div>
        <span
          className="MuiTouchRipple-root-85"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/forum/321"
    >
      <div
        className="MuiButtonBase-root-43 MuiListItem-root-33 MuiListItem-default-36 MuiListItem-gutters-40 MuiListItem-button-41"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiAvatar-root-46 MuiAvatar-colorDefault-47"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-49"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </g>
          </svg>
        </div>
        <div
          className="MuiListItemText-root-55"
        >
          <h3
            className="MuiTypography-root-61 MuiTypography-subheading-68 MuiListItemText-primary-58"
          >
            Bar subject
          </h3>
          <p
            className="MuiTypography-root-61 MuiTypography-body1-70 MuiTypography-colorTextSecondary-83 MuiListItemText-secondary-59"
          >
            Criado por Bar Foo h치 segundos atr치s
          </p>
        </div>
        <span
          className="MuiTouchRipple-root-85"
        />
      </div>
    </div>
  </ul>
</div>
`;
